grpcutil/naming.go

```diff
@@ -26,11 +26,11 @@ type Update struct {
 
 // Resolver creates a Watcher for a target to track its resolution changes.
 type Resolver interface {
-	// Resolve creates a Watcher for target.
-	Resolve(target string) (Watcher, error)
+	// Resolve creates a Watcher for target/service.
```

Comment:
Why do we even *need* this interface in the first place? It is not *used* in dskit, so it should not exist here. We should export the implementation instead. (same for `Watcher`). That's probably a separate PR.

Reply:
@pstibrany Let's do it in a separate PR then?

Reply:
Yes. Let's see what others think too.

Reply:
@pstibrany https://github.com/grafana/dskit/pull/105

I realized that the reason for the `Watcher` interface is that there are two different implementations: `dnsWatcher` and `ipWatcher`. I.e., getting rid if the interface is trickier. Do you still think we should do it?